graphql_compiler/compiler/workarounds/orientdb_query_execution.py

```diff
@@ -267,8 +267,10 @@ def _expose_only_preferred_locations(match_query, location_types, coerced_locati
                     # we ensure that we again infer the same type bound.
                     eligible_location_types[current_step_location] = current_type_bound
 
-                    if current_step_location not in coerced_locations:
-                        # The type bound here is already implied by the GraphQL query structure.
+                    if (current_step_location not in coerced_locations or
+                            previous_type_bound is not None):
```

Comment:
Use `any` here instead of multi-line `if`?

Reply:
I think @lodrion's preference might be the other way. If I'm wrong, I'll fix it in the next PR.